vllm/v1/core/scheduler.py

```diff
@@ -182,6 +237,13 @@ def schedule(self) -> "SchedulerOutput":
                     self.encoder_cache_manager.allocate(request, i)
                 encoder_budget = new_encoder_budget
 
+            # Track if we need guided decoding
+            # Create individual bitmask for requests with grammar
+            if request.grammar is not None:
+                if request.request_id not in guided_decoding_bitmasks:
+                    bitmask = request.grammar.allocate_bitmask(1, vocab_size)
+                    guided_decoding_bitmasks[request.request_id] = bitmask
```

Comment:
Is it possible to put this bitmask in each request's data instead of returning a single dictionary that holds all these bitmasks for each request? Naively it looks like these could be put onto the instances of each `Grammar` object on each request- i.e. `request.grammar.bistmask`, but I assume there's a reason they're not?

Reply:
+1, the bitmask should be attached to the request if possible